### PR TITLE
update animal ai

### PR DESCRIPTION
**Pull request by HentaiStorm**
## About The Pull Request

upgrade superior animal AI

## Why It's Good For The Game

Mobs will look for new nearby targets when receiving damage

## Changelog
:cl:
tweak: superior animal ai
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/superior_animal/defense.dm [hunk 1]
```diff
@@ -185,6 +186,11 @@
 	if (overkill_dust && (amount >= overkill_dust) && (getFireLoss() >= maxHealth*2))
 		dust()
 
+/mob/living/carbon/superior_animal/proc/reagr_new_targets(var/reagr_radius = 1))
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/carbon/superior_animal/defense.dm:189]:
```suggestion
/mob/living/carbon/superior_animal/proc/reagr_new_targets(reagr_radius = 1)
```

**Comment by MLGTASTICa** [comment]
Why is this needed ? the AI works fine most of the time. At most this change might be laggy when you have 10 roaches on a tile fighting a spider and them constantly cycling eachother by looping through view and through the mobs nearby needlessly .

**Comment by HentaiStorm** [comment]
> 
> 
> Why is this needed ? the AI works fine most of the time. At most this change might be laggy when you have 10 roaches on a tile fighting a spider and them constantly cycling eachother by looping through view and through the mobs nearby needlessly .

There is a point when the mob pursues the target and does not pay attention to the attacks in the back. This is fix mob.

**Comment by MLGTASTICa** [comment]
> HOSTILE_STANCE_ATTACK - This is moving to target, not attacking. If the mob is already attacking someone, the reagr will not execute.
> 
> https://github.com/discordia-space/CEV-Eris/blob/68d2bbbbab3bb4231f8501e73130f29b0b14be66/code/modules/mob/living/carbon/superior_animal/superior_animal.dm#L278

it still goes through everything in view(1) and calls find_target for how many superior animals are there. target_mob is also.. the target , not just moving. For what the performance impact could be i don't see why this is needed and if the behaviour it provides is of any use at all.

**Review by TheShown911 — APPROVED** [review]
This should work then the current system, might be problems with mobs re-aggro-ing on different targets in the same fight though.

**Pull request merged**